runtime/vm.go

```diff
@@ -1236,6 +1329,33 @@ mainloop:
 		}
 	}
 
+	if err != nil {
+		for v.ip < v.curIPLimit && (atomic.LoadInt64(&v.aborting) == 0) {
+			if v.framesIndex <= 1 {
+				break
+			}
```

Comment:
can be `for v.ip < v.curIPLimit && v.framesIndex > 1 {`? No need to check `aborting` as the main loop has already finished.

Reply:
Haha yes, absolutely. I guess I was in a hurry to check the optimisation and forgot to clean it up. I'll do the cleanups and push in the morning.

Reply:
It's no problem. Hey, can you add some tests for modules? Basically module imports are compiled into functions and it should work seamlessly, I guess.

Also, I tend to squash all commits when merging the PRs. So my recommendation is to create another PR on `master` after I merge mine if you want to preserve your commits. (Also I might want to merge mine sooner and don't want to rush you.)

Reply:
Sure